iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/serviceclient/RegistryManagerTests.java

```diff
@@ -456,35 +456,76 @@ public void apply_configuration_e2e() throws Exception
-    public void deviceCreationWithSecurityScope() throws IOException, InterruptedException, IotHubException, URISyntaxException
+    public void deviceCreationWithDeviceScope() throws IOException, InterruptedException, IotHubException, URISyntaxException
     {
         // Arrange
         RegistryManagerTestInstance testInstance = new RegistryManagerTestInstance();
         deleteDeviceIfItExistsAlready(testInstance.registryManager, testInstance.deviceId);
 
         //-Create-//
-        Device edgeDevice = Device.createFromId(testInstance.deviceId, DeviceStatus.Enabled, null);
+        Device edgeDevice1 = Device.createFromId(testInstance.deviceId, DeviceStatus.Enabled, null);
         DeviceCapabilities capabilities = new DeviceCapabilities();
         capabilities.setIotEdge(true);
-        edgeDevice.setCapabilities(capabilities);
-        edgeDevice = Tools.addDeviceWithRetry(testInstance.registryManager, edgeDevice);
+        edgeDevice1.setCapabilities(capabilities);
+        edgeDevice1 = Tools.addDeviceWithRetry(testInstance.registryManager, edgeDevice1);
 
-        Device leafDevice = Device.createFromId(testInstance.deviceId + "-leaf", DeviceStatus.Enabled, null);
-        assertNotNull(edgeDevice.getScope());
-        leafDevice.setScope(edgeDevice.getScope());
+        Device edgeDevice2 = Device.createFromId(testInstance.deviceId, DeviceStatus.Enabled, null);
+        capabilities.setIotEdge(true);
+        edgeDevice2.setCapabilities(capabilities);
+        edgeDevice2.getParentScopes().add(edgeDevice1.getScope()); // set edge1 as parent
+        edgeDevice2 = Tools.addDeviceWithRetry(testInstance.registryManager, edgeDevice2);
+
+        Device leafDevice = Device.createFromId(
+                testInstance.deviceId + "-leaf",
+                DeviceStatus.Enabled,
+                null);
+        assertNotNull(edgeDevice1.getScope());
+        leafDevice.setScope(edgeDevice1.getScope());
         Tools.addDeviceWithRetry(testInstance.registryManager, leafDevice);
 
-
         //-Read-//
         Device deviceRetrieved = testInstance.registryManager.getDevice(testInstance.deviceId);
 
         //-Delete-//
         testInstance.registryManager.removeDevice(testInstance.deviceId);
 
         // Assert
-        assertEquals(buildExceptionMessage("Registered device id is not correct", hostName), testInstance.deviceId, edgeDevice.getDeviceId());
-        assertEquals(buildExceptionMessage("Registered device id is not correct", hostName), testInstance.deviceId, deviceRetrieved.getDeviceId());
-        assertEquals(buildExceptionMessage("Security scopes did not match", hostName), deviceRetrieved.getScope(), edgeDevice.getScope());
+        assertEquals(
+                buildExceptionMessage(
+                        "Edge parent scope did not match parent's device scope",
+                        hostName),
+                edgeDevice2.getParentScopes().indexOf(0),
+                edgeDevice1.getScope());
+        assertNotEquals(
```

Comment:
Confirmed that edge device scope does not change when it has a parent (unlike a leaf device, which does not have its own device scope).